services/builder/configProcessor.go

```diff
@@ -68,66 +49,115 @@ func configProcessor(c pb.ContextBoxServiceClient, wg *sync.WaitGroup) error {
-					desiredCluster:       tmpDesired,
-					loadbalancers:        clusterView.Loadbalancers[clusterName],
-					desiredLoadbalancers: clusterView.DesiredLoadbalancers[clusterName],
-					deletedLoadBalancers: clusterView.DeletedLoadbalancers[clusterName],
+					desiredCluster:       clusterView.CurrentClusters[clusterName],
+					desiredLoadbalancers: clusterView.Loadbalancers[clusterName],
 					Workflow:             clusterView.ClusterWorkflows[clusterName],
 				}
 
-				if ctx, err = buildCluster(ctx, c); err != nil {
+				if err := callKubeEleven(ctx, c); err != nil {
 					clusterView.SetWorkflowError(clusterName, err)
 					log.Error().Msgf("Failed to build cluster %s project %s : %v", clusterName, config.Name, err)
 					return err
 				}
-				log.Info().Msgf("First stage for cluster %s project %s finished building", clusterName, config.Name)
 
-				// make the desired state of the temporary cluster the new current state.
 				clusterView.CurrentClusters[clusterName] = ctx.desiredCluster
 				clusterView.Loadbalancers[clusterName] = ctx.desiredLoadbalancers
+
+				if err := callPatchClusterInfoConfigMap(ctx, c); err != nil {
+					clusterView.SetWorkflowError(clusterName, err)
+					log.Error().Msgf("Failed to build cluster %s project %s : %v", clusterName, config.Name, err)
+					return err
+				}
 			}
 
-			if toDelete != nil {
+			if len(diff.ToDelete) > 0 {
+				currentStage++
+				clusterView.ClusterWorkflows[clusterName].Description = fmt.Sprintf("Processing stage [%d/%d]", currentStage, stages)
+				log.Info().Msgf("Processing stage [%d/%d] for cluster %s config %s", currentStage, stages, clusterName, config.Name)
+
 				clusterView.ClusterWorkflows[clusterName].Stage = pb.Workflow_DELETE_NODES
 				if err := updateWorkflowStateInDB(config.Name, clusterName, clusterView.ClusterWorkflows[clusterName], c); err != nil {
 					clusterView.SetWorkflowError(clusterName, err)
 					return err
 				}
 				log.Info().Msgf("Deleting nodes from cluster %s project %s", clusterName, config.Name)
-				if clusterView.CurrentClusters[clusterName], err = deleteNodes(clusterView.CurrentClusters[clusterName], toDelete); err != nil {
+				if clusterView.CurrentClusters[clusterName], err = deleteNodes(clusterView.CurrentClusters[clusterName], diff.ToDelete); err != nil {
 					clusterView.SetWorkflowError(clusterName, err)
 					log.Error().Msgf("Failed to delete nodes cluster %s project %s : %v", clusterName, config.Name, err)
 					return err
 				}
 			}
 
 			message := fmt.Sprintf("Processing cluster %s config %s", clusterName, config.Name)
-			if tmpDesired != nil {
-				clusterView.ClusterWorkflows[clusterName].Description = "Processing stage [2/2]"
-				message = fmt.Sprintf("Processing stage [2/2] for cluster %s config %s", clusterName, config.Name)
+			if diff.Stages() > 0 {
+				currentStage++
+				clusterView.ClusterWorkflows[clusterName].Description = fmt.Sprintf("Processing stage [%d/%d]", currentStage, stages)
```

Comment:
Just a clarification, now, there can be up to 3 stages?

1. Adding new nodes (just k8s, LB is ignored)
2. Deleting nodes
3. Changing API EP and adding LBs?

Reply:
Yes.

1. is the `adding && deleting` stage or now also `adding & adding LB api endpoint` with the tmp config.
2. the api endpoint change if needed
3. the deletion of nodes if needed

then the original workflow.